File: coding-for-fun/go/list_files.go
// author: https://github.com/AnotherGithubDude, 2022
// tested with: go1.19.4
package main

import (
	"fmt"
	"log"
        "io/ioutil"
	"sort"
	"time"
)

func main() {

	// Print the current time in hours, minutes, seconds
	t := time.Now()
	fmt.Println("The current time is:", t.Format("3:04:05 PM"))

	// List the files of the current folder, each per line, sorted alphabetically
	files, err := ioutil.ReadDir("./")
	if err != nil {
		log.Fatal(err)
	}
	sort.Slice(files, func(i, j int) bool {
		return files[i].Name() < files[j].Name()
	})
	for _, file := range files {
		fmt.Println(file.Name())
	}
}

File: coding-for-fun/python/list_files.py
# author: https://github.com/AnotherGithubDude, 2022
# tested with: Python 3.10.8
import os
import time
from pathlib import Path

def main():
	# Print the current time in hours, minutes, seconds
	t = time.localtime()
	print("The current time is:", time.strftime("%I:%M:%S %p", t))

	# List the files of the current folder, each per line, sorted alphabetically
	files = os.listdir(Path("."))
	files.sort()
	for file in files:
		print(file)

if __name__ == "__main__":
    main()
